ivadomed/testing.py

```diff
@@ -410,6 +410,9 @@ def get_gt(filenames):
     Returns:
         ndarray: 4D numpy array.
     """
+    # Check filenames extentions and update paths if not NifTI
+    filenames = [imed_loader_utils.update_filename_to_nifti(fname) for fname in filenames]
+
```

Comment:
I tried it earlier this week and it's a whole other can of worms. 
So it is possible but it also requires several changes to the code that were not better in my opinion.

1. First step would be returning and updating the filenames with the function `read_file` [here](https://github.com/ivadomed/ivadomed/blob/9e3dfdacf2c38bf452a51991ed7385ae4c897043/ivadomed/loader/loader.py#L209), [here](https://github.com/ivadomed/ivadomed/blob/9e3dfdacf2c38bf452a51991ed7385ae4c897043/ivadomed/loader/loader.py#L224) and [here](https://github.com/ivadomed/ivadomed/blob/9e3dfdacf2c38bf452a51991ed7385ae4c897043/ivadomed/loader/loader.py#L226). Then updating the `self.metadata` [here](https://github.com/ivadomed/ivadomed/blob/9e3dfdacf2c38bf452a51991ed7385ae4c897043/ivadomed/loader/loader.py#L251) with the new filenames. That would work for `train` and `test` without any changes in `testing.py` and `inference.py`, but would require changes to `evalution.py` (point 2).

2. For `evaluation.py`, the solution would remain the same as the one I implemented because the filenames are taken [here](https://github.com/ivadomed/ivadomed/blob/9e3dfdacf2c38bf452a51991ed7385ae4c897043/ivadomed/evaluation.py#L53) from the `bids_dataframe` which does not have the updated filenames. We do not want to update the `bids_dataframe` to keep a  reference to the original files and metadata.

3.  For the `segment` command and through the `segment_volume` (sct_deepseg) entry point, we would need changes to `inference.py`. We would have to fetch the new filenames from the metadata in the DataLoader batch (ex: `batch['input_metadata'][0][0]['input_filenames']`) somewhere around [here](https://github.com/ivadomed/ivadomed/blob/9e3dfdacf2c38bf452a51991ed7385ae4c897043/ivadomed/inference.py#L497) to pass to `pred_to_nib`, instead of using the filename list from the parameters.

So my conclusion after all of this was that updating the filenames was not worth it, it seemed to me that this solution was a lot harder to implement, understand and maintain.  Moreover, there would be discrepancies between the paths, filenames and extensions in the metadata  unless we update them all.
Again, let me know what you think, thanks!

(and now that I look at it again, I should probably use my `update_filename_to_nifti` function instead of the `filename.split('.')[0] + ".nii.gz"`).

Reply:
Thanks for being so thorough about this! I'm convinced. :slightly_smiling_face:

Reply:
> (and now that I look at it again, I should probably use my update_filename_to_nifti function instead of the filename.split('.')[0] + ".nii.gz").

Fixed